### ISSUE TITLE
View codes and logic codes should not be coupled

### ISSUE THREAD

**Issue by cychiuae** [nodegin/lihkg-web#10]
The codes for fetching post should be separated from component codes

**Comment by jackytck** [comment]
Yes, e.g. in Thread.js, it should be pure and should not reloadPosts by itself. It should dispatch an action to reload posts, then await the result in redux-thunk, ... etc

**Comment by chewong** [comment]
I can refactor it over the christmas break
